File: backend/src/main/java/com/vehicle/maintenance/dto/maintenance_plan/VehicleMaintenancePlanMatchResponse.java
package com.vehicle.maintenance.dto.maintenance_plan;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

/**
 * Respuesta de resolución vehículo → plan preventivo cargado manualmente por MMY,
 * opcionalmente fijado en el vehículo (mantenimiento_plan_id).
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class VehicleMaintenancePlanMatchResponse {

    private Long vehicleId;
    private String vehicleBrand;
    private String vehicleModel;
    private Integer vehicleYear;
    private Integer vehicleMileage;
    private String licensePlate;

    /** FIXED = plan manual en vehículo; MMY = emparejó marca/modelo/año; NONE = sin plan */
    private String matchMode;

    /** ID del plan usado cuando matchMode FIXED o MMY */
    private Long resolvedPlanId;

    private MaintenancePlanResponse plan;

    /** Texto aclaratorio para el taller (ej. cargar plan o revisar marca/modelo). */
    private String hint;

    /**
     * IDs de {@code maintenance_plan_activities} excluídos sólo para esta placa.
     */
    @Builder.Default
    private List<Long> excludedPreventiveActivityIds = new ArrayList<>();

    private Integer lastServiceMileage;
    private java.time.LocalDate lastServiceDate;

    /** Próximo servicio acordado en ficha (único por vehículo; km y/o fecha). */
    private Integer nextCommittedServiceMileage;
    private java.time.LocalDate nextCommittedServiceDate;
}

File: app_mantenimiento/lib/models/maintenance_plan/vehicle_maintenance_plan_match_response.dart
import 'maintenance_plan_response.dart';

class VehicleMaintenancePlanMatchResponse {
  final int vehicleId;
  final String? vehicleBrand;
  final String? vehicleModel;
  final int? vehicleYear;
  final int? vehicleMileage;
  final String? licensePlate;
  final String matchMode;
  final int? resolvedPlanId;
  final MaintenancePlanResponse? plan;
  final String? hint;
  final int? lastServiceMileage;
  final String? lastServiceDate;
  final int? nextCommittedServiceMileage;
  final String? nextCommittedServiceDate;

  VehicleMaintenancePlanMatchResponse({
    required this.vehicleId,
    this.vehicleBrand,
    this.vehicleModel,
    this.vehicleYear,
    this.vehicleMileage,
    this.licensePlate,
    this.matchMode = 'NONE',
    this.resolvedPlanId,
    this.plan,
    this.hint,
    this.lastServiceMileage,
    this.lastServiceDate,
    this.nextCommittedServiceMileage,
    this.nextCommittedServiceDate,
  });

  factory VehicleMaintenancePlanMatchResponse.fromJson(
      Map<String, dynamic> json) {
    MaintenancePlanResponse? p;
    if (json['plan'] != null &&
        json['plan'] is Map<String, dynamic>) {
      p = MaintenancePlanResponse.fromJson(
          Map<String, dynamic>.from(json['plan'] as Map));
    }

    return VehicleMaintenancePlanMatchResponse(
      vehicleId: (json['vehicleId'] as num?)?.toInt() ?? 0,
      vehicleBrand: json['vehicleBrand']?.toString(),
      vehicleModel: json['vehicleModel']?.toString(),
      vehicleYear: (json['vehicleYear'] as num?)?.toInt(),
      vehicleMileage: (json['vehicleMileage'] as num?)?.toInt(),
      licensePlate: json['licensePlate']?.toString(),
      matchMode: json['matchMode']?.toString() ?? 'NONE',
      resolvedPlanId: (json['resolvedPlanId'] as num?)?.toInt(),
      plan: p,
      hint: json['hint']?.toString(),
      lastServiceMileage:
          (json['lastServiceMileage'] as num?)?.toInt(),
      lastServiceDate: json['lastServiceDate']?.toString(),
      nextCommittedServiceMileage:
          (json['nextCommittedServiceMileage'] as num?)?.toInt(),
      nextCommittedServiceDate:
          json['nextCommittedServiceDate']?.toString(),
    );
  }
}
